superset/utils/log.py

```diff
@@ -72,32 +100,25 @@ def log(  # pylint: disable=too-many-arguments
     ) -> None:
         pass
 
-    @contextmanager
-    def log_context(  # pylint: disable=too-many-locals
-        self, action: str, object_ref: Optional[str] = None, log_to_statsd: bool = True,
-    ) -> Iterator[Callable[..., None]]:
-        """
-        Log an event with additional information from the request context.
-
-        :param action: a name to identify the event
-        :param object_ref: reference to the Python object that triggered this action
-        :param log_to_statsd: whether to update statsd counter for the action
-        """
+    def log_with_context(
+        self,
+        action: str,
+        duration: timedelta,
+        object_ref: Optional[str] = None,
+        log_to_statsd: bool = True,
+        payload_override: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
```

Comment:
I think it would be nice to have `payload_override` be passed like we're doing in the context manager:

```suggestion
        **payload_override: Optional[Dict[str, Any]],
```

This way if we don't want to use the context manager we can do:

```python
event_logger.log_with_context(action="foo", engine="bar")
```